pkg/gateway/gateway.go

```diff
@@ -228,3 +224,133 @@ func (gw *Gateway) GetNetwork(name string) (*Network, error) {
 func (gw *Gateway) Close() {
 	// future use
 }
+
+func createGatewayConfigProvider(config core.ConfigProvider, gw *Gateway) func() ([]core.ConfigBackend, error) {
+	return func() ([]core.ConfigBackend, error) {
+		configBackend, err := config()
+		if err != nil {
+			return nil, err
+		}
+		if len(configBackend) != 1 {
+			return nil, errors.New("invalid config file")
+		}
+
+		cfg := configBackend[0]
+
+		lhConfig := make([]core.ConfigBackend, 0)
+		lhConfig = append(lhConfig, createGatewayConfig(cfg, gw))
+
+		return lhConfig, nil
+	}
+}
+
+func createGatewayConfig(backend core.ConfigBackend, gw *Gateway) *gatewayConfig {
+	var matchers map[string][]map[string]string
+	if strings.ToUpper(os.Getenv("DISCOVERY_AS_LOCALHOST")) == "TRUE" {
```

Comment:
you may want to set a const for the env variable name.

Reply:
done